### ISSUE TITLE
undefined method `commit' for nil:NilClass

### ISSUE THREAD

**Issue by greyltc** [gollum/gollum#1530]
Hi, I just followed the guide here: https://github.com/gollum/gollum/wiki/Gollum-via-Docker but it didn't work.
When I navigate to the supposed wiki page in my browser I get an error page in my browser:
```
NoMethodError at /Home
undefined method `commit' for nil:NilClass
```
The container's command line pukes like this:
```
$ docker run -v `pwd`:/wiki -p 4567:80 gollum
/usr/local/bundle/gems/sinatra-contrib-2.0.8.1/lib/sinatra/namespace.rb:335: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1348: warning: The called method `before' is defined here
/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1526: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:26: warning: The called method `run' is defined here
[2020-03-31 00:03:21] INFO  WEBrick 1.6.0
[2020-03-31 00:03:21] INFO  ruby 2.7.0 (2019-12-25) [x86_64-linux]
== Sinatra (v2.0.8.1) has taken the stage on 80 for production with backup from WEBrick
[2020-03-31 00:03:21] INFO  WEBrick::HTTPServer#start: pid=1 port=80
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
172.17.0.1 - - [31/Mar/2020:00:04:50 UTC] "GET / HTTP/1.1" 302 0
- -> /
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
2020-03-31 00:04:50 - NoMethodError - undefined method `commit' for nil:NilClass:
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/redirects.rb:15:in `init'
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/wiki.rb:548:in `redirects'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:563:in `show_page_or_file'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:539:in `block in <class:App>'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `block in compile!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (3 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1006:in `route_eval'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (2 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1035:in `block in process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:985:in `block in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `each'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1098:in `block in dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1095:in `dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `block in call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:908:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/xss_header.rb:18:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/path_traversal.rb:16:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/json_csrf.rb:26:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/frame_options.rb:31:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/null_logger.rb:11:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/head.rb:12:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/show_exceptions.rb:22:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:194:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1951:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `block in call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1730:in `synchronize'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:95:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:140:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:96:in `run'
	/usr/local/lib/ruby/2.7.0/webrick/server.rb:307:in `block in start_thread'
172.17.0.1 - - [31/Mar/2020:00:04:50 UTC] "GET /Home HTTP/1.1" 500 131752
- -> /Home
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
2020-03-31 00:04:50 - NoMethodError - undefined method `commit' for nil:NilClass:
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/redirects.rb:15:in `init'
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/wiki.rb:548:in `redirects'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:563:in `show_page_or_file'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:539:in `block in <class:App>'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `block in compile!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (3 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1006:in `route_eval'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (2 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1035:in `block in process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:985:in `block in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `each'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1098:in `block in dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1095:in `dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `block in call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:908:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/xss_header.rb:18:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/path_traversal.rb:16:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/json_csrf.rb:26:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/frame_options.rb:31:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/null_logger.rb:11:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/head.rb:12:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/show_exceptions.rb:22:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:194:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1951:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `block in call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1730:in `synchronize'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:95:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:140:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:96:in `run'
	/usr/local/lib/ruby/2.7.0/webrick/server.rb:307:in `block in start_thread'
172.17.0.1 - - [31/Mar/2020:00:04:50 UTC] "GET /__sinatra__/500.png HTTP/1.1" 500 131231
http://127.0.0.1:4567/Home -> /__sinatra__/500.png
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
2020-03-31 00:04:51 - NoMethodError - undefined method `commit' for nil:NilClass:
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/redirects.rb:15:in `init'
	/usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/wiki.rb:548:in `redirects'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:563:in `show_page_or_file'
	/usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:539:in `block in <class:App>'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `block in compile!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (3 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1006:in `route_eval'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (2 levels) in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1035:in `block in process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `process_route'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:985:in `block in route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `each'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `route!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1098:in `block in dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1095:in `dispatch!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `block in call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `call!'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:908:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/xss_header.rb:18:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/path_traversal.rb:16:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/json_csrf.rb:26:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
	/usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/frame_options.rb:31:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/null_logger.rb:11:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/head.rb:12:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/show_exceptions.rb:22:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:194:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1951:in `call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `block in call'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1730:in `synchronize'
	/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `call'
	/usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:95:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:140:in `service'
	/usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:96:in `run'
	/usr/local/lib/ruby/2.7.0/webrick/server.rb:307:in `block in start_thread'
172.17.0.1 - - [31/Mar/2020:00:04:51 UTC] "GET /favicon.ico HTTP/1.1" 500 131153
http://127.0.0.1:4567/Home -> /favicon.ico

```
I'll attach the html error page shown in my browser here 
[NoMethodError at _Home.zip](https://github.com/gollum/gollum/files/4406039/NoMethodError.at._Home.zip)

What should I do to fix that?

**Comment by dometto** [comment]
@greyltc thanks for the report. This is a bug in gollum-lib that is unrelated to Docker; it happens when using a completely new repository (one that doesn't even have a HEAD commit yet). I'll get to fixing it now, and you'll be able to fix the big installing the latest `gollum-lib` (which I guess will happen automatically in your Docker setup).

**Comment by sberlin** [comment]
@dometto Could you please check, if the following error is related? Executed with Docker and `gollum:5.0.0` based on Alpine image. All of my pages are broken, even after executing the migration script. 
```
/usr/local/bundle/gems/sinatra-contrib-2.0.8.1/lib/sinatra/namespace.rb:335: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1348: warning: The called method `before' is defined here
/usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1526: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:26: warning: The called method `run' is defined here
[2020-03-31 09:20:03] INFO  WEBrick 1.6.0
[2020-03-31 09:20:03] INFO  ruby 2.7.0 (2019-12-25) [x86_64-linux-musl]
== Sinatra (v2.0.8.1) has taken the stage on 8080 for production with backup from WEBrick
[2020-03-31 09:20:03] INFO  WEBrick::HTTPServer#start: pid=1 port=8080
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
2020-03-31 09:20:07 - NoMethodError - undefined method `[]' for nil:NilClass:
        /usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/markup.rb:107:in `skip_filter?'
        /usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/markup.rb:157:in `block in render'
        /usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/markup.rb:157:in `reject'
        /usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/markup.rb:157:in `render'
        /usr/local/bundle/gems/gollum-lib-5.0.1/lib/gollum-lib/page.rb:168:in `formatted_data'
        /usr/local/bundle/gems/gollum-5.0.0/lib/gollum/views/page.rb:91:in `has_header'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:129:in `[]'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:129:in `find'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:99:in `block in fetch'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:95:in `each'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:95:in `fetch'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:77:in `[]'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/template.rb:290:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/template.rb:43:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache.rb:125:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/context.rb:34:in `partial'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/template.rb:60:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/template.rb:43:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache.rb:121:in `render'
        /usr/local/bundle/gems/mustache-0.99.8/lib/mustache/sinatra.rb:113:in `mustache'
        /usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:560:in `show_page_or_file'
        /usr/local/bundle/gems/gollum-5.0.0/lib/gollum/app.rb:539:in `block in <class:App>'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `block in compile!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (3 levels) in route!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1006:in `route_eval'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (2 levels) in route!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1035:in `block in process_route'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `catch'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `process_route'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:985:in `block in route!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `each'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `route!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1098:in `block in dispatch!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1095:in `dispatch!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `block in call!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `call!'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:908:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/xss_header.rb:18:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/path_traversal.rb:16:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/json_csrf.rb:26:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
        /usr/local/bundle/gems/rack-protection-2.0.8.1/lib/rack/protection/frame_options.rb:31:in `call'
        /usr/local/bundle/gems/rack-2.2.2/lib/rack/null_logger.rb:11:in `call'
        /usr/local/bundle/gems/rack-2.2.2/lib/rack/head.rb:12:in `call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/show_exceptions.rb:22:in `call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:194:in `call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1951:in `call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `block in call'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1730:in `synchronize'
        /usr/local/bundle/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `call'
        /usr/local/bundle/gems/rack-2.2.2/lib/rack/handler/webrick.rb:95:in `service'
        /usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:140:in `service'
        /usr/local/lib/ruby/2.7.0/webrick/httpserver.rb:96:in `run'
        /usr/local/lib/ruby/2.7.0/webrick/server.rb:307:in `block in start_thread'
172.17.0.1 - - [31/Mar/2020:09:20:07 UTC] "GET /Home HTTP/1.1" 500 178609
- -> /Home
/usr/local/bundle/gems/gollum-rugged_adapter-0.99.4/lib/rugged_adapter/git_layer_rugged.rb:562: warning: Using the last argument as keyword parameters is deprecated
172.17.0.1 - - [31/Mar/2020:09:20:08 UTC] "GET /__sinatra__/500.png HTTP/1.1" 404 1310
http://localhost:8080/Home -> /__sinatra__/500.png
```

**Comment by dometto** [comment]
@sberlin no, if you're running this on an existing repo, it's not related. Please open a new issue, and provide the output of `gollum --versions`, and preferably an example repo.

@greyltc note that in the mean time you could get around this by adding a commit to the repo via the command line.

**Comment by greyltc** [comment]
Great, thanks for the workaround. I'll give that a go now...

**Comment by dometto** [comment]
@sberlin could it be that you are using a `config.rb` file with customized formats?

**Comment by sberlin** [comment]
@dometto No, but while setting up a repo to reproduce I discovered our `_Header.html` seems to be the cause of the problem. Now I'm looking up custom HTML integration in gollum and if anything changed in the behavior.

**Comment by dometto** [comment]
@greyltc this issue has been fixed in gollum-lib (https://github.com/gollum/gollum-lib/pull/370). I'm just waiting to do a bugfix release because maybe @sberlin's issue will also require a fix to gollum-lib.

**Comment by sberlin** [comment]
@dometto Any hints on why including `_Header.html` was possible prior to 5.0.0? The wiki [suggests](https://github.com/gollum/gollum/wiki#subpages) only the documented extensions are considered for templating. HTML ist not among them.

**Comment by dometto** [comment]
> @dometto Any hints on why including _Header.html was possible prior to 5.0.0? The wiki suggests only the documented extensions are considered for templating. HTML ist not among them.

I've never seen a case of gollum being used with `.html` pages or subpages. Perhaps it was working 'accidentally', but there certainly aren't any tests in gollum-lib for this behavior, nor any documentation. Or perhaps your organisation was using a custom `config.rb` or `config.ru` that was overriding gollum's markup settings (see below)?

You could:

* Switch to using markdown for your header, sidebar, footer
* If you want html in the header because you want to change the layout of the wiki, you could instead try one of these options: https://github.com/gollum/gollum/wiki/Custom-Styling-and-Views
* You could try to hack in support for HTML pages by registering HTML as a custom markup. Something like this in a `config.rb` might work:

```ruby
module Gollum
  class Markup
    register(:html, "html", :skip_filters => [:Tags, :Code, :Render], :enabled => true)
  end
end
```

The idea would be to skip filters modifying the content of the `.html` file so the raw HTML from the file is inserted .

**Comment by dometto** [comment]
@greyltc's issue has been fixed in `gollum-lib v5.0.2`. Closing this. @sberlin if you have further questions or issues, please open a new issue.
